CMakeLists.txt

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright(c) 2006 to 2019 ADLINK Technology Limited and others
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+# v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#
+cmake_minimum_required(VERSION 3.7)
+
+# Set a default build type if none was specified
+set(default_build_type "RelWithDebInfo")
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
+      STRING "Choose the type of build." FORCE)
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+    "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+# By default the Java-based components get built, but make it possible to disable that: if only the
+# core library is required, there's no need to build them, and that in turn eliminates the Maven and
+# JDK dependency.
+option(BUILD_IDLC "Build IDL preprocessor" ON)
+option(BUILD_CONFTOOL "Build configuration file edit tool" ON)
+
+# By default don't treat warnings as errors, else anyone building it with a different compiler that
+# just happens to generate a warning, as well as anyone adding or modifying something and making a
+# small mistake would run into errors.  CI builds can be configured differently.
+option(WERROR "Treat compiler warnings as errors" OFF)
+
+FUNCTION(PREPEND var prefix)
+   SET(listVar "")
+   FOREACH(f ${ARGN})
+      LIST(APPEND listVar "${prefix}/${f}")
+   ENDFOREACH(f)
+   SET(${var} "${listVar}" PARENT_SCOPE)
+ENDFUNCTION(PREPEND)
+
+# Update the git submodules
+execute_process(COMMAND git submodule init
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND git submodule update
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+
+# Set module path before defining project so platform files will work.
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake/Modules")
+set(CMAKE_PROJECT_NAME_FULL "Eclipse Cyclone DDS")
+set(PROJECT_NAME "CycloneDDS")
+project(${PROJECT_NAME} VERSION 0.1.0)
```

Comment:
I would suggest to not use a variable here. You can just write `project(CycloneDDS VERSION 0.1.0)` and the CMake variable `PROJECT_NAME` will be set by CMake automatically.

What is the advantage of this? By looking into the CMake code and finding the `project()` call the project name can be extracted since it is a string literal (e.g. by tools like `colcon`).

Reply:
I created #235 doing that.